components/user_settings/display/manage_languages/manage_languages.jsx

```diff
@@ -21,15 +24,57 @@ export default class ManageLanguage extends React.Component {
 
     constructor(props) {
         super(props);
+        const locales = I18n.getLanguages();
+        const userLocale = props.locale;
+        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        this.reactSelect = React.createRef();
+        this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
+            selectedOption,
             isSaving: false,
         };
     }
 
-    setLanguage = (e) => {
-        this.setState({locale: e.target.value});
+    componentDidMount() {
+        if (this.reactSelectContainer.current) {
+            const reactSelectInput = this.reactSelectContainer.current.querySelector('input');
+            reactSelectInput.addEventListener('keydown', this.handleKeyDown);
```

Comment:
If you're going to listen to input of ReactSelect, you may pass a function into its `onKeydown`.  If on container, then directly on container like:
```javascript
this.reactSelectContainer.current.addEventListener('keydown', this.handleContainerKeyDown);
```